### PR TITLE
feat(sca): move SCA models from ggshield to pygitguardian

### PR DESCRIPTION
**Pull request by xblanchot-gg**

### CODE REVIEW

File: pygitguardian/client.py [hunk 1]
```diff
@@ -605,3 +611,90 @@ def create_jwt(
...
+        return result
+
+    def sca_scan_directory(
+        self,
+        tar_file: bytes,
+        scan_parameters: SCAScanParameters,
+        extra_headers: Optional[Dict[str, str]] = None,
+    ) -> Union[Detail, SCAScanAllOutput]:
+        """
+        Generates tar archive associated with filenames and launches
```
Comment (agateau-gg) [inline pygitguardian/client.py:646]: +1 for adding a docstring, but from what I understand, the function does not *generates* a tar archive, it expects the bytes of a tar archive to be passed to it, isn't it?

File: tests/test_sca_models.py [hunk 2]
```diff
@@ -0,0 +1,194 @@
...
+                },
+            ),
+        ],
+    )
+    def test_schema_loads(self, expected_klass, instance_data):
+        """
+        GIVEN the right kwargs and an extra field in dict format
+        WHEN loading using the schema
+        THEN the extra field is not taken into account
+            AND the result should be an instance of the expected class
```
Comment (agateau-gg) [inline tests/test_sca_models.py:187]: nit: it's odd to indent `AND` (I see it's been done in two other tests in this repo, I think it's a mistake)

```suggestion
        AND the result should be an instance of the expected class
```

**Review by agateau-gg — CHANGES_REQUESTED** [review]
Minor remarks, looks good overall.

File: pygitguardian/client.py (hunk 1 above)
Comment (xblanchot-gg) [reply pygitguardian/client.py:646]: You're right, that seems to be outdated ! Corrected

File: tests/test_sca_models.py (hunk 2 above)
Comment (xblanchot-gg) [reply tests/test_sca_models.py:187]: Indeed, the mistake has been done once and copy/paste everywhere. fixed

**Review by agateau-gg — APPROVED** [review]

**Pull request merged**